backvendor/gosource.go

```diff
@@ -57,34 +57,39 @@ type GoSource struct {
 	usesGodep bool
 }
 
-func findExcludes(pth string, globs []string) (map[string]struct{}, error) {
-	excludes := make(map[string]struct{})
+// FindExcludes returns a slice of paths which match the provided
+// globs.
+func FindExcludes(pth string, globs []string) ([]string, error) {
+	excludes := make([]string, 0)
 	for _, glob := range globs {
 		matches, err := filepath.Glob(filepath.Join(pth, glob))
 		if err != nil {
 			return nil, err
 		}
 		for _, match := range matches {
-			excludes[match] = struct{}{}
+			excludes = append(excludes, match)
 		}
 	}
 	return excludes, nil
 }
 
-func NewGoSource(pth string, excludeGlobs ...string) (*GoSource, error) {
-	excludes, err := findExcludes(pth, excludeGlobs)
-	if err != nil {
-		return nil, err
+// NewGoSource returns a *GoSource for the given path pth. The paths
+// in excludes will not be considered when matching against the
+// upstream repository.
+func NewGoSource(pth string, excludes ...string) (*GoSource, error) {
```

Comment:
Why is `excludes` a variadic parameter when you already have a slice of strings returned from FindExcludes to pass in? Is this a best practice?

Reply:
Maybe a slice would read a little better actually.

Reply:
Changed this to a slice.